tsl/src/continuous_aggs/utils.c

```diff
@@ -707,3 +708,237 @@ continuous_agg_migrate_to_time_bucket(PG_FUNCTION_ARGS)
+	Assert(view_name != NULL);
+	namestrcpy(&view_name_name, view_name);
+
+	got_next_slot = index_getnext_slot(indexscan, ForwardScanDirection, slot);
+	Ensure(!got_next_slot, "found duplicate definitions for CAgg mat_ht %d", mat_hypertable_id);
+
+	/* End relation scan */
+	index_endscan(indexscan);
+	ExecDropSingleTupleTableSlot(slot);
+	relation_close(cagg_rel, AccessShareLock);
+	relation_close(cagg_idx_rel, AccessShareLock);
+
+	/* Get Oid of user view */
+	Oid direct_view_oid =
+		ts_get_relation_relid(NameStr(view_schema_name), NameStr(view_name_name), false);
+	Assert(OidIsValid(direct_view_oid));
+	return direct_view_oid;
+}
+
+enum
+{
+	Anum_cagg_bucket_function_oid = 1,
+	Anum_cagg_bucket_function_width,
+	Anum_cagg_bucket_function_origin,
+	Anum_cagg_bucket_function_offset,
+	Anum_cagg_bucket_function_timezone,
+	Anum_cagg_bucket_function_fixed_width,
+	_Anum_cagg_bucket_function_max
+};
+
+#define Natts_cagg_bucket_function (_Anum_cagg_bucket_function_max - 1)
+
+static Datum
+create_cagg_get_bucket_function_datum(TupleDesc tupdesc, ContinuousAggsBucketFunction *bf)
+{
+	NullableDatum datums[Natts_cagg_bucket_function] = { { 0 } };
+	HeapTuple tuple;
+
+	char *bucket_origin = NULL;
+	char *bucket_offset = NULL;
+	char *bucket_width = NULL;
+
+	if (IS_TIME_BUCKET_INFO_TIME_BASED(bf))
+	{
+		/* Bucketing on time */
+		Assert(bf->bucket_time_width != NULL);
+		bucket_width = DatumGetCString(
+			DirectFunctionCall1(interval_out, IntervalPGetDatum(bf->bucket_time_width)));
+
+		if (!TIMESTAMP_NOT_FINITE(bf->bucket_time_origin))
+		{
+			bucket_origin = DatumGetCString(
+				DirectFunctionCall1(timestamptz_out, TimestampTzGetDatum(bf->bucket_time_origin)));
+		}
+
+		if (bf->bucket_time_offset != NULL)
+		{
+			bucket_offset = DatumGetCString(
+				DirectFunctionCall1(interval_out, IntervalPGetDatum(bf->bucket_time_offset)));
+		}
+	}
+	else
+	{
+		/* Bucketing on integers */
+		bucket_width = palloc0(MAXINT8LEN + 1 * sizeof(char));
+		pg_lltoa(bf->bucket_integer_width, bucket_width);
+
+		/* Integer buckets with origin are not supported, so noting to do. */
+		Assert(bucket_origin == NULL);
+
+		if (bf->bucket_integer_offset != 0)
+		{
+			bucket_offset = palloc0(MAXINT8LEN + 1 * sizeof(char));
+			pg_lltoa(bf->bucket_integer_offset, bucket_offset);
+		}
+	}
+
+	tupdesc = BlessTupleDesc(tupdesc);
+
+	/* Bucket function */
+	ts_datum_set_objectid(Anum_cagg_bucket_function_oid, datums, bf->bucket_function);
+
+	/* Bucket width */
+	ts_datum_set_text_from_cstring(Anum_cagg_bucket_function_width, datums, bucket_width);
+
+	/* Bucket origin */
+	ts_datum_set_text_from_cstring(Anum_cagg_bucket_function_origin, datums, bucket_origin);
+
+	/* Bucket offset */
+	ts_datum_set_text_from_cstring(Anum_cagg_bucket_function_offset, datums, bucket_offset);
+
+	/* Bucket timezone */
+	ts_datum_set_text_from_cstring(Anum_cagg_bucket_function_timezone,
+								   datums,
+								   bf->bucket_time_timezone);
```

Comment:
Nit: it's obvious from the attribute names what these are so comments are not needed.

Reply:
Fixed!